client/Commands/Prompts/RollDicePrompt.tsx

```diff
@@ -1,30 +1,40 @@
 import { Field } from "formik";
 import React = require("react");
+import { StatBlock } from "../../../common/StatBlock";
 import { probablyUniqueString } from "../../../common/Toolbox";
 import { SubmitButton } from "../../Components/Button";
-import { Dice } from "../../Rules/Dice";
-import { RollResult } from "../../Rules/RollResult";
+import { Formula } from "../../Rules/Formulas/Formula";
+import { FormulaResult } from "../../Rules/Formulas/FormulaTerm";
 import { PromptProps } from "./components/PendingPrompts";
 
 interface RollDiceModel {
   diceExpression: string;
 }
 
 export const RollDicePrompt = (
-  rollDiceExpression: (expression: string) => void
+  displayRollResult: (expression: string, result: FormulaResult) => boolean,
+  selectedStatBlock: () => StatBlock
 ): PromptProps<RollDiceModel> => {
   const fieldLabelId = probablyUniqueString();
+  let errors: any = {};
```

Comment:
use Formik error handling or extract as a class component to use setState() for error messaging.